### PR TITLE
Support Different Storage Targets for KubeVirt Provider 

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
Currently, machine controller only supports the creation of KubeVirt Data Volumes using PVCs without specifying what storage profile will be used which is quite important for some CSI drivers. 
**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind feature

**Special notes for your reviewer**:
This PR was tested manually with KubeVirt and two machine have been created and they were reflected correctly on the VMs: 
```shell
spec:
  dataVolumeTemplates:
  - metadata:
      creationTimestamp: null
      name: dv-test-557f6b6c88-2z7g2
    spec:
      source:
        http:
          url: http://remote/host/os
      storage:
        accessModes:
        - ReadWriteOnce
        resources:
          requests:
            storage: 25Gi
        storageClassName: nfs
```
**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
support `storage ` as a another KubeVirt DataVolume storage definition 
```

**Documentation**:

```documentation
None
```

### CODE REVIEW

File: pkg/cloudprovider/provider/kubevirt/types/types.go [hunk 1]
```diff
@@ -76,6 +76,9 @@ type Template struct {
 // PrimaryDisk.
 type PrimaryDisk struct {
 	Disk
+	// StorageTarget describes which VirtualMachine storage target will be used in the DataVolumeTemplate.
+	StorageTarget providerconfigtypes.ConfigVarString `json:"storageTarget,omitempty"`
```
Comment (yaa110) [inline pkg/cloudprovider/provider/kubevirt/types/types.go:80]: please add some tests in `provider_test.go`

**Review by yaa110 — CHANGES_REQUESTED** [review]

File: pkg/cloudprovider/provider/kubevirt/types/types.go (hunk 1 above)
Comment (moadqassem) [reply pkg/cloudprovider/provider/kubevirt/types/types.go:80]: Done https://github.com/kubermatic/machine-controller/pull/1855/commits/40c283c65384a8482ff2a3c18832997fd1d96c32

**Review by yaa110 — APPROVED** [review]

**Pull request merged**